include/qore/QoreNamespace.h

```diff
@@ -309,4 +358,130 @@ class QoreNamespaceConstIterator {
     DLLEXPORT const QoreNamespace* get() const;
 };
 
+//! allows local namespaces to be iterated
+/** @since %Qore 0.8.13
+ */
+class QoreNamespaceNamespaceIterator {
+public:
+    //! creates the iterator
+    DLLEXPORT QoreNamespaceNamespaceIterator(const QoreNamespace* ns);
+
+    //! destroys the iterator
+    DLLEXPORT ~QoreNamespaceNamespaceIterator();
+
+    //! moves to the next position; returns true if on a valid position
+    DLLEXPORT bool next();
+
+    //! returns the namespace
+    DLLEXPORT const QoreNamespace* get() const;
+
+private:
+    class qore_namespace_namespace_iterator* priv;
```

Comment:
why the PImpl idiom is used here?
also, let's use std::unique_ptr instead of explicitly allocating and deleting the pointer.